includes/class-api.php

```diff
@@ -358,6 +361,13 @@ public function attach_featured_image( $post_id, $post_args ) {
 
 		$response = set_post_thumbnail( $post_id, $thumbnail_id );
 
+		// Remove filter that allowed download_url() to use an external host.
+		remove_filter( 'http_request_host_is_external', array( $this, 'allow_sync_external_host' ) );
+
+	}
+
+	public function allow_sync_external_host( $allow, $host, $url ) {
```

Comment:
This filter usually returns false but we need it to return true only while we are attempting to download the featured image. After the image has been downloaded then we can return it to its previous setting.